terraform/modules/rds/main.tf

```
@@ -0,0 +1,56 @@
+# Non-compliant RDS configuration (missing encryption, backups, and monitoring)
+resource "aws_db_instance" "main" {
+  identifier        = "${var.environment}-db"
```

The RDS instance is not compliant with SOC2 rules due to missing encryption, insufficient backup retention, and lack of monitoring. These are critical for ensuring data security and availability.

```suggestion:-0+0
resource "aws_db_instance" "main" {
  identifier        = "${var.environment}-db"
  engine            = "postgres"
  engine_version    = "13.7"
  instance_class    = var.instance_class
  allocated_storage = 20

  name              = var.db_name
  username          = var.db_username
  password          = var.db_password

  publicly_accessible = false

  storage_encrypted = true

  backup_retention_period = 30
  skip_final_snapshot     = false

  monitoring_interval = 60
  performance_insights_enabled = true

  vpc_security_group_ids = [aws_security_group.db.id]

  tags = {
    Name        = "${var.environment}-db"
    Environment = var.environment
  }
}
```